doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,93 @@
+and how the dispatching mechanism in scikit-image works.
+
+.. important::
+    Dispatching support and infrastructure is experimental and is not ready for production.
+    It's made available as an early draft so that developers can gain experience
+    with the system.
+
+    Expect the dispatching API to change without notice.
+
+
+Implementing a new backend
+--------------------------
+
+An alternative implementation ("backend") is a good place to provide optimized implementations
+for particular hardware, support array libraries other than Numpy or explore novel ideas that
+are not (yet) a good fit for the core scikit-image library.
+
+To create a backend you have to create a new Python package that registers two particular
+`entrypoints <https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points>`_. The first entrypoint called `skimage_backends` should resolve to a namespace
+that contains two functions: `can_has(name, *args, **kwargs)` and `get_implementation(name)`.
+The second entrypoint called `skimage_backend_infos` should resolve to a function with no
+arguments that returns an instance of the `skimage.util._backends.BackendInformation` class
+or something that behaves like it.
+
+
+The `skimage_backend_infos` entrypoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The information your backend provides via this entrypoint is used to get things like its
+name, homepage and which functions it implements. The main requirement for this entrypoint
+is that it is fast to import and has no dependencies other than `scikit-image`.
+
+The return value of the function should be an instance of the
+`skimage.util._backends.BackendInformation` class or something that behaves like it.
+
+The reason this entrypoint has to be fast is that the list of implemented functions which
+it provides is used to make a decision on which scikit-image functions need decorating
+with dispatching logic.
+
+To help make your implementation fast avoid computing the list of implemented functions
+dynamically or performing other expensive operations.
+
+
+The `skimage_backends` entrypoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This entrypoint should point to a namespace that contains two functions:
+`can_has(name, *args, **kwargs)` and `get_implementation(name)`. They will be used to
+determine if the backend will be called and to get the actual implementation.
+
+When a user calls a function that the backend listed as one that it implements (via
+the information endpoint) the
+`can_has` function will be called with the name of the function that is being dispatched
+and the arguments the user provided when calling the function. The `can_has` function
+should use this information to determine if the backend wants to be called or if a
+different backend should be tried. A backend might implement a particular function but
+only want to handle calls where the input arrays are of a particuler type or size.
+
```

Comment:
Leaving this out for now as it is not implemented.

Reply:
We don't have to implement it. It works because we are not checking with which arguments we are dispatching the call to the backend function. see here - https://github.com/Schefflera-Arboricola/skimage-j4f/blob/main/skimage_j4f/metrics/simple_metrics.py#L6